encoding/marshal_test.go

```diff
@@ -216,18 +216,58 @@ func TestEncodeDecode(t *testing.T) {
 	}
 }
 
+func TestEncodeAll(t *testing.T) {
+	var expected []byte
+	for i := range testStructs {
+		expected = append(expected, Marshal(testStructs[i])...)
+	}
```

Comment:
This test depends on the correctness of `Marshal`, I think would be better avoided, at least there should be 1 or 2 tests where you manually create structs and hardcode the expected output. It's worth having both, but I've found that it's worth having hardcoded tests.

Even something as simple as `EncodeAll(uint64(12), [][2]byte{{1,2},{3,4}}) => [hardcoded result]` would make me feel better.